components/eventing-controller/pkg/subscriptionmanager/jetstream/jetstream.go

```diff
@@ -102,9 +107,62 @@ func (sm *SubscriptionManager) Start(defaultSubsConfig env.DefaultSubscriptionCo
 	return nil
 }
 
-func (sm *SubscriptionManager) Stop(_ bool) error {
+func (sm *SubscriptionManager) Stop(runCleanup bool) error {
 	sm.cancel()
-	sm.namedLogger().Info("stopped JetStream subscription manager")
+	if !runCleanup {
+		return nil
+	}
+	dynamicClient := dynamic.NewForConfigOrDie(sm.restCfg)
+	return cleanup(sm.backend, dynamicClient, sm.namedLogger())
+}
+
+// clean removes all JetStream artifacts.
+func cleanup(backend handlers.JetStreamBackend, dynamicClient dynamic.Interface, logger *zap.SugaredLogger) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var ok bool
+	var jsBackend *handlers.JetStream
+	if jsBackend, ok = backend.(*handlers.JetStream); !ok {
+		err := errors.New("convert backend handler to JetStream handler failed")
+		logger.Errorw("no JetStream backend exists", "error", err)
+		return err
+	}
+
+	// fetch all subscriptions.
+	subscriptionsUnstructured, err := dynamicClient.Resource(handlers.SubscriptionGroupVersionResource()).Namespace(corev1.NamespaceAll).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "list subscriptions failed")
+	}
+
+	subs, err := handlers.ToSubscriptionList(subscriptionsUnstructured)
+	if err != nil {
+		return errors.Wrapf(err, "convert subscriptionList from unstructured list failed")
+	}
+
+	// clean all status.
+	isCleanupSuccessful := true
+	for _, v := range subs.Items {
+		sub := v
+		subKey := types.NamespacedName{Namespace: sub.Namespace, Name: sub.Name}
+		log := logger.With("key", subKey.String())
+
+		desiredSub := handlers.RemoveStatus(sub)
```

Comment:
out-of-scope since you did not introduce it: I think `RemoveStatus` is the wrong name because it is actually overriding the status with the defaults. 

Maybe like this ? 🤔 

```suggestion
		desiredSub := handlers.ResetStatusToDefaults(sub)
```

Reply:
It's actually removing the Status and replacing it with an empty Subscription Status which has just the Ready field. I don't think defaults is the right term, but I can't think of anything better. So I'll take the suggestion 👍

Reply:
Mhh, I know what you mean 🤔. Defaulting is more the defaulting in Go which is associated with non pointer types.